clients/java/src/test/java/io/camunda/zeebe/client/resource/rest/DeployResourceRestTest.java

```diff
@@ -0,0 +1,473 @@
+        .hasUrl(RestGatewayPaths.getDeploymentsUrl());
+  }
+
+  @Test
+  public void shouldDeployResourceFromBytes() {
+    // given
+    final String filename = BPMN_1_FILENAME;
+    final byte[] bytes = getBytes(filename);
+
+    // when
+    client.newDeployResourceCommand().addResourceBytes(bytes, filename).send().join();
+
+    // then
+    LoggedRequestAssert.assertThat(RestGatewayService.getLastRequest())
+        .hasMethod(RequestMethod.POST)
+        .hasUrl(RestGatewayPaths.getDeploymentsUrl());
+  }
+
+  @Test
+  public void shouldDeployResourceFromString() {
+    // given
+    final String filename = BPMN_1_FILENAME;
+    final String xml = new String(getBytes(filename));
+
+    // when
+    client
+        .newDeployResourceCommand()
+        .addResourceString(xml, StandardCharsets.UTF_8, filename)
+        .send()
+        .join();
+
+    // then
+    LoggedRequestAssert.assertThat(RestGatewayService.getLastRequest())
+        .hasMethod(RequestMethod.POST)
+        .hasUrl(RestGatewayPaths.getDeploymentsUrl());
+  }
+
+  @Test
+  public void shouldDeployResourceFromUtf8String() {
+    // given
+    final String filename = BPMN_1_FILENAME;
+    final String xml = new String(getBytes(filename), StandardCharsets.UTF_8);
+
+    // when
+    client.newDeployResourceCommand().addResourceStringUtf8(xml, filename).send().join();
+
+    // then
+    LoggedRequestAssert.assertThat(RestGatewayService.getLastRequest())
+        .hasMethod(RequestMethod.POST)
+        .hasUrl(RestGatewayPaths.getDeploymentsUrl());
+  }
+
+  @Test
+  public void shouldDeployResourceFromProcessModel() {
+    // given
+    final String filename = "test.bpmn";
+    final BpmnModelInstance processModel =
+        Bpmn.createExecutableProcess(BPMN_1_PROCESS_ID).startEvent().endEvent().done();
+
+    final ByteArrayOutputStream outStream = new ByteArrayOutputStream();
+    Bpmn.writeModelToStream(outStream, processModel);
+    final byte[] expectedBytes = outStream.toByteArray();
+
+    // when
+    client.newDeployResourceCommand().addProcessModel(processModel, filename).send().join();
+
+    // then
+    LoggedRequestAssert.assertThat(RestGatewayService.getLastRequest())
+        .hasMethod(RequestMethod.POST)
+        .hasUrl(RestGatewayPaths.getDeploymentsUrl());
+  }
+
+  @Test
+  public void shouldDeployMultipleResources() {
+    // given
+    final long key = 345L;
+    final String tenantId = "test-tenant";
+    final String filename1 = BPMN_1_FILENAME.substring(1);
+    final String filename2 = BPMN_2_FILENAME.substring(1);
+
+    // when
+    client
+        .newDeployResourceCommand()
+        .addResourceFromClasspath(filename1)
+        .addResourceFromClasspath(filename2)
+        .tenantId(tenantId)
+        .send()
+        .join();
+
+    // then
+    LoggedRequestAssert.assertThat(RestGatewayService.getLastRequest())
+        .hasMethod(RequestMethod.POST)
+        .hasUrl(RestGatewayPaths.getDeploymentsUrl());
+  }
```

Comment:
❓ Are we able to assert the resource data bytes that we send from the client like we do for gRPC?

Reply:
I'll have a look

Reply:
Synced offline:
* The gRPC request simply holds the request data separately which is why it's easily accessible there for tests.
* In the REST request, this is encoded inside the multipart form data request, which makes it less accessible in the tests.
* We'll create a follow-up to improve the tests. We're confident that with integration tests that use the client and verify the content arrives well we're well covered for now nonetheless.